include/llbuild/BuildSystem/BuildValue.h

```diff
@@ -193,9 +205,11 @@ class BuildValue {
 
   bool isMissingOutput() const { return kind == Kind::MissingOutput; }
   bool isFailedInput() const { return kind == Kind::FailedInput; }
+  bool isCancelledInput() const { return kind == Kind::CancelledInput; }
```

Comment:
Is there a difference between a cancelled input and a failed input? Both seem like failure, or do we need to retain both cases?